### PR TITLE
Allow to pass a blob to core fetch function

### PR DESCRIPTION
**Pull request by cyriltovena**
**What is this feature?**

This allows to pass a Blob and not try to parse it when using the core fetch function which is used by plugins when communicating with backend service.

**Why do we need this feature?**

I have feature that needs to send protobuf encoded data and this is blocking.

**Who is this feature for?**

Mostly for developer using fetch.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/core/utils/fetch.ts [hunk 1]
```diff
@@ -89,6 +90,9 @@ export const parseBody = (options: BackendSrvRequest, isAppJson: boolean) => {
   if (!options.data || typeof options.data === 'string') {
     return options.data;
   }
+  if (options.data.constructor.name === 'Blob') {
```
Comment (joshhunt) [inline public/app/core/utils/fetch.ts:93]: does `options.data instanceof Blob` work here?

Comment (cyriltovena) [reply public/app/core/utils/fetch.ts:93]: No apparently jest need a polyfill but I'm not sure how to do it.

Comment (aocenas) [reply public/app/core/utils/fetch.ts:93]: @joshhunt should we add a polyfill for this? see https://github.com/jsdom/jsdom/issues/2555 and https://github.com/elastic/kibana/pull/162197

Comment (aocenas) [reply public/app/core/utils/fetch.ts:93]: Ok seems like there actually is a Blob polyfill https://github.com/grafana/grafana/blob/e7797ac4395b431715810315c3137b9e1913f509/public/test/jest-setup.ts#L9.

Comment (cyriltovena) [reply public/app/core/utils/fetch.ts:93]: I wasn't sure how to use it properly.

File: public/app/core/utils/fetch.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { Blob } from 'buffer';
```
Comment (aocenas) [inline public/app/core/utils/fetch.ts:1]: Ah ok this seems to be the problem. This is buffer from the node API which we (probably) don't want to use. We want to use the browser Blob (which is available globally) or use the polyfill in Jest. With this import you created a "native" Blob in the test but then were comparing it with the node Blob imported here (as that shadowed the global one).

Comment (cyriltovena) [reply public/app/core/utils/fetch.ts:1]: honestly I wasn't able to make it work

**Review by aocenas — APPROVED** [review]

**Pull request merged**